app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,133 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+class HomeViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeScreenMapper: HomeScreenMapper,
+) : ViewModel() {
+    private val popularCategories = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES
+    )
+
+    private val nowPlayingCategories = listOf(
+        MovieCategory.NOW_PLAYING_MOVIES,
+        MovieCategory.NOW_PLAYING_TV
+    )
+
+    private val upcomingCategories = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK
+    )
+
+    private val popularSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.POPULAR_STREAMING)
+    private val upcomingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.UPCOMING_TODAY)
+    private val nowPlayingSelectedCategory: MutableState<MovieCategory> =
+        mutableStateOf(MovieCategory.NOW_PLAYING_TV)
+
+    private val _popularMoviesHomeViewState = MutableStateFlow(HomeMovieCategoryViewState.EMPTY)
+    val popularMoviesHomeViewState: StateFlow<HomeMovieCategoryViewState> =
+        _popularMoviesHomeViewState.asStateFlow()
```

Comment:
You should implement this the same way you did FavoritesViewState in FavoritesViewModel. That way you won't be needing private `_VS` you will just have popularMoviewHomeViewState. Same goes for nowPlaying and upcoming.